core/src/main/scala/cats/data/NonEmptySet.scala

```diff
@@ -0,0 +1,218 @@
+    new SemigroupK[NonEmptySet] with Reducible[NonEmptySet] {
+
+      def combineK[A](a: NonEmptySet[A], b: NonEmptySet[A]): NonEmptySet[A] =
+        a ++ b
+
+      override def size[A](fa: NonEmptySet[A]): Long = fa.length.toLong
+
+      override def reduceLeft[A](fa: NonEmptySet[A])(f: (A, A) => A): A =
+        fa.reduceLeft(f)
+
+      override def reduce[A](fa: NonEmptySet[A])(implicit A: Semigroup[A]): A =
+        fa.reduce
+
+      def reduceLeftTo[A, B](fa: NonEmptySet[A])(f: A => B)(g: (B, A) => B): B = fa.reduceLeftTo(f)(g)
+
+      def reduceRightTo[A, B](fa: NonEmptySet[A])(f: A => B)(g: (A, Eval[B]) => Eval[B]): Eval[B] =
+        fa.reduceRightTo(f)(g)
+
+      override def foldLeft[A, B](fa: NonEmptySet[A], b: B)(f: (B, A) => B): B =
+        fa.foldLeft(b)(f)
+
+      override def foldRight[A, B](fa: NonEmptySet[A], lb: Eval[B])(f: (A, Eval[B]) => Eval[B]): Eval[B] =
+        fa.foldRight(lb)(f)
+
+      override def foldMap[A, B](fa: NonEmptySet[A])(f: A => B)(implicit B: Monoid[B]): B =
+        B.combineAll(fa.toSet.iterator.map(f))
+
+      override def fold[A](fa: NonEmptySet[A])(implicit A: Monoid[A]): A =
+        fa.reduce
+
+      override def find[A](fa: NonEmptySet[A])(f: A => Boolean): Option[A] =
+        fa.find(f)
+
+      override def forall[A](fa: NonEmptySet[A])(p: A => Boolean): Boolean =
+        fa.forall(p)
+
+      override def exists[A](fa: NonEmptySet[A])(p: A => Boolean): Boolean =
+        fa.exists(p)
+
+      override def toList[A](fa: NonEmptySet[A]): List[A] = fa.toSet.toList
+
+      override def toNonEmptyList[A](fa: NonEmptySet[A]): NonEmptyList[A] =
+        NonEmptyList(fa.head, fa.tail.toList)
+    }
+
+  implicit def catsDataEqForNonEmptySet[A: Order]: Eq[NonEmptySet[A]] =
+    new Eq[NonEmptySet[A]]{
+      def eqv(x: NonEmptySet[A], y: NonEmptySet[A]): Boolean = x === y
+    }
+
+  implicit def catsDataShowForNonEmptySet[A](implicit A: Show[A]): Show[NonEmptySet[A]] =
+    Show.show[NonEmptySet[A]](_.show)
+
+  implicit def catsDataBandForNonEmptySet[A]: Band[NonEmptySet[A]] = new Band[NonEmptySet[A]] {
+    def combine(x: NonEmptySet[A], y: NonEmptySet[A]): NonEmptySet[A] = x ++ y
+  }
+}
+
+object NonEmptySet extends NonEmptySetInstances {
+  def fromSet[A: Order](as: SortedSet[A]): Option[NonEmptySet[A]] =
+    if (as.nonEmpty) Option(new NonEmptySet(as)) else None
+
+  def fromSetUnsafe[A: Order](set: SortedSet[A]): NonEmptySet[A] =
+    if (set.nonEmpty) new NonEmptySet(set)
+    else throw new IllegalArgumentException("Cannot create NonEmptySet from empty set")
+
+
+  def of[A: Order](a: A, as: A*): NonEmptySet[A] =
+    new NonEmptySet(SortedSet(a)(Order[A].toOrdering) ++ SortedSet(as: _*)(Order[A].toOrdering) + a)
+  def apply[A: Order](head: A, tail: SortedSet[A]): NonEmptySet[A] = new NonEmptySet(SortedSet(head)(Order[A].toOrdering) ++ tail)
+  def one[A: Order](a: A): NonEmptySet[A] = new NonEmptySet(SortedSet(a)(Order[A].toOrdering))
```

Comment:
do we need this one vs `of`? I also like `singleton` or something better.

Reply:
I took the name from `NonEmptyList.one`, but I have no preference wrt naming.

Reply:
I vote for consistent API